### ISSUE TITLE
[a11y]: Modal: Footer buttons not fully in viewport at 400% zoom

### ISSUE THREAD

**Issue by gcheang** [carbon-design-system/carbon#19139]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.80.0

### Automated testing tool and ruleset

n/a

### Description

At 400% zoom, longer modal heading texts can wrap multiple lines and push the footer buttons outside of the viewport. In extreme cases, the buttons are pushed completely outside of the viewport.

### WCAG 2.1 Violation

1.4.10 Reflow

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?globals=&args=&id=components-modal--default&viewMode=story

### Steps to reproduce

1. Change zoom level to 400%.
2. Edit the modal heading text until it wraps twice or more.
3. See footer buttons are no longer fully in the viewport. In the example below, button labels are now hidden.

Example screenshot:
<img width="1552" alt="Image" src="https://github.com/user-attachments/assets/033319f0-16ff-4e44-bf5f-b85124a985ad" />

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by annawen1** [comment]
Hi @gcheang I'm not seeing this issue occurring, can you provide more details?

This is testing with Chrome Version 135.0.7049.96
https://github.com/user-attachments/assets/ff0b1839-2f5b-4f02-9112-0eaba987fb4c

**Comment by gcheang** [comment]
Hi @annawen1, please edit the heading text to something longer, such as in my example screenshot. The problem is exacerbated the longer the heading text is and the more times it wraps.

**Comment by annawen1** [comment]
@gcheang Ah my apologies, I missed that step!

**Comment by preetibansalui** [comment]
Hey, I fixed it by adding a max height to the modal header. But just a heads-up — it’s best to keep modal titles short, as suggested in the [Carbon docs](https://carbondesignsystem.com/components/modal/usage/#title).

> Use a brief verb phrase that describes the modal’s task or purpose.
